### PR TITLE
feat: Added support for `--azure-credential` via `VSCE_AZURE_CREDENTIAL` environment variable

### LINKED ISSUE #662: Add support for VSCE authentication via the `--azure-credential` flag
Hard to find in the documentation, but the flag can be found here: https://github.com/microsoft/vscode-vsce/blob/cf0bc3242b9adac3b25c9d7ea0de78842203d123/src/main.ts#L191

I might take a swing at doing this at some point as Microsoft leans away from PATs.

### PR DESCRIPTION
**Pull request by JakeShirley**
vsce added support for `--azure-credential` a few months ago as Microsoft moves away from long-living PATs.  This flag allows authentication to be pulled from the current azure cli context.

This PR also does the following small tweaks:
- Adds a `launch.json` to run ava tests for debugging via VS Code
- Fixes the test `publishes an extension in a non-root folder` to work on Windows (some hard-coded pathing assumptions were made)
- Adds a `.gitattributes` file to force line endings to be `LF` instead of `CRLF`.  On Windows prettier was complaining about all of the line endings, this change makes git clone/treat all of the files as `LF`.

Closes #662

### CODE REVIEW

File: lib/verify.js [hunk 1]
```diff
@@ -16,15 +16,15 @@ module.exports = async (pluginConfig, { logger, cwd }) => {
     const ovsxPublishEnabled = isOvsxPublishEnabled();
     if (!vscePublishEnabled && !ovsxPublishEnabled) {
       throw new SemanticReleaseError(
-        'No personal access token was detected. Set the `VSCE_PAT` or the `OVSX_PAT` environment variable, at least one of them must be present when publish is enabled.\nLearn more at https://github.com/felipecrs/semantic-release-vsce#publishing',
+        'No personal access token was detected. Set the `VSCE_PAT`, `VSCE_USE_AZURE_CREDENTIALS`, or the `OVSX_PAT` environment variable, at least one of them must be present when publish is enabled.\nLearn more at https://github.com/felipecrs/semantic-release-vsce#publishing',
```
Comment (felipecrs) [inline lib/verify.js]: I think VSCE_AZURE_CREDENTIALS would look better.

Also, please update the documentation (README).

**Comment by JakeShirley** [comment]
Validated this with my fork. Working for me! :) 

https://github.com/Mojang/minecraft-debugger/actions/runs/10120996092/job/27991427579

**Comment by felipecrs** [comment]
@JakeShirley, thanks a lot. I'm currently on vacation far away from the computer.  I should review your PR next week.

**Comment by JakeShirley** [comment]
@felipecrs did you ever get a chance to look at this? :)

File: README.md [hunk 2]
```diff
@@ -96,11 +96,12 @@ The directory of the extension relative to the current working directory. Defaul
...
+| `OVSX_PAT`               | _Optional_. The personal access token to push to Open VSX Registry                                                                                                                                                                                                                                              |
+| `VSCE_PAT`               | _Optional_. The personal access token to publish to Visual Studio Marketplace. _Note:_ Cannot be set at the same time as `VSCE_AZURE_CREDENTIALS`.                                                                                                                                                              |
+| `VSCE_AZURE_CREDENTIALS` | _Optional_. When set `vsce` will use the `--azure-credential` flag to authenticate. _Note:_ Cannot be set at the same time as `VSCE_PAT`                                                                                                                                                                        |
```
Comment (felipecrs) [inline README.md]:
```suggestion
| `VSCE_AZURE_CREDENTIALS` | _Optional_. When set to any value, `vsce` will use the `--azure-credential` flag to authenticate. _Note:_ Cannot be set at the same time as `VSCE_PAT`.                                                                                                                                                                        |
```

File: README.md [hunk 3]
```diff
@@ -171,8 +172,10 @@ jobs:
       - run: npx semantic-release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          # In case you want to publish to Visual Studio Marketplace
+          # In case you want to publish to Visual Studio Marketplace.  Remove if using VSCE_AZURE_CREDENTIALS
           VSCE_PAT: ${{ secrets.VSCE_PAT }}
+          # Set in order to publish to Visual Studio Marketplace using Azure CLI credentials
+          #VSCE_AZURE_CREDENTIALS: true
```
Comment (felipecrs) [inline README.md]: I'd rather omit this. If users want to use it, they will know how to use it. There's even your example below.

What do you think?

File: lib/utils.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ const isOvsxPublishEnabled = () => {
 };
 
 const isVscePublishEnabled = () => {
-  return 'VSCE_PAT' in process.env;
+  return 'VSCE_PAT' in process.env || 'VSCE_AZURE_CREDENTIALS' in process.env;
```
Comment (felipecrs) [inline lib/utils.js]: I've got to say I'm tempted to say VSCE_USE_AZURE_CREDENTIALS would be a better name, but it's not a strong opinion.

Do you have one?

File: lib/verify-vsce-auth.js [hunk 5]
```diff
@@ -5,16 +5,29 @@ const execa = require('execa');
 
 module.exports = async (logger, cwd) => {
   logger.log('Verifying authentication for vsce');
+  logger.log(`VSCE_AZURE_CREDENTIALS: ${process.env.VSCE_AZURE_CREDENTIALS}`);
```
Comment (felipecrs) [inline lib/verify-vsce-auth.js]: this is more like debug. Please remove it

File: lib/verify-vsce-auth.js [hunk 6]
```diff
@@ -5,16 +5,29 @@ const execa = require('execa');
...
 
-  if (!process.env.VSCE_PAT) {
+  if (!process.env.VSCE_PAT && !process.env.VSCE_AZURE_CREDENTIALS) {
     throw new SemanticReleaseError(
-      'Empty vsce personal access token (`VSCE_PAT` environment variable) specified.',
-      'EEMPTYVSCEPAT',
+      'Neither vsce personal access token (`VSCE_PAT` environment variable) or Azure Credentials flag (`VSCE_AZURE_CREDENTIALS` environment variable) specified.',
+      'EVSCEAUTHNOTPROVIDED',
     );
   }
```
Comment (felipecrs) [inline lib/verify-vsce-auth.js]: You'll have to change this a little. This actually checks whether the variable is present but empty, like: VSCE_PAT="".

Maybe you should invent two cases and two error codes? One to check for empty vsce pat and another to check for empty VSCE_AZURE_CREDENTIALS.

if ('VSCE_PAT' in process.env && !process.env.VSCE_PAT) {
  // old condition
}

// something similar for AZURE_CREDENTIALS

File: lib/verify-vsce-auth.js [hunk 7]
```diff
@@ -5,16 +5,29 @@ const execa = require('execa');
...
 
+  if (process.env.VSCE_PAT && process.env.VSCE_AZURE_CREDENTIALS) {
```
Comment (felipecrs) [inline lib/verify-vsce-auth.js]:
```suggestion
  if ('VSCE_PAT' in process.env && 'VSCE_AZURE_CREDENTIALS' in process.env) {
```

And I think this check should come before the previous check, don't you think?

File: lib/verify-vsce-auth.js [hunk 8]
```diff
@@ -5,16 +5,29 @@ const execa = require('execa');
...
+    throw new SemanticReleaseError(
+      'Both vsce personal access token (`VSCE_PAT` environment variable) or Azure Credentials flag (`VSCE_AZURE_CREDENTIALS` environment variable) specified.  Please use only one.',
+      'EVSCEDUPLICATEAUTHPROVIDED',
+    );
+  }
+
+  const vsceFlags = ['verify-pat'];
```
Comment (felipecrs) [inline lib/verify-vsce-auth.js]:
```suggestion
  const vsceArgs = ['verify-pat'];
```

Or just args.

File: package.json [hunk 9]
```diff
@@ -48,7 +48,7 @@
   },
   "dependencies": {
     "@semantic-release/error": "^3.0.0",
-    "@vscode/vsce": "^2.15.0",
+    "@vscode/vsce": "^2.31.1",
```
Comment (felipecrs) [inline package.json]: I'll update vsce in a separate PR. You can keep this here while I do that, though.

**Review by felipecrs — CHANGES_REQUESTED** [review]

**Comment by felipecrs** [comment]
> @felipecrs did you ever get a chance to look at this? :)

Very sorry for taking this long. I had actually forgot.

File: package.json (hunk 9 above)
Comment (felipecrs) [reply package.json]: It's done, you can rebase now.

File: lib/utils.js (hunk 4 above)
Comment (felipecrs) [reply lib/utils.js]: or actually, we should name it VSCE_AZURE_CREDENTIAL, to align with the term in vsce (--azure-credential).

**Comment by JakeShirley** [comment]
@felipecrs Finally got around to addressing this feedback :)

**Comment by JakeShirley** [comment]
Did a fresh rebase, hopefully all of the checks are good now :)

**Review by felipecrs — APPROVED** [review]

**Pull request merged**